src/samplers/mcmc/mcmc_state.jl

```diff
@@ -301,27 +301,13 @@ function mcmc_update_z_position!!(mcmc_state::MCMCState)
 end
 
 function mcmc_update_z_position!!(mc_state::MCMCChainState)
-    f_transform = mc_state.f_transform
+    f_inv = inverse(mc_state.f_transform)
 
-    x_current = mc_state.current.x.v
-    logd_x_current = mc_state.current.x.logd
+    current_z_new = transform_samples(f_inv, mc_state.current.x)
+    proposed_z_new = transform_samples(f_inv, mc_state.proposed.x)
```

Comment:
To ensure we're type-stable:

```suggestion
    current_z_new::typeof(mc_state.current.z) = transform_samples(f_inv, mc_state.current.x)
    proposed_z_new::typeof(mc_state_new.proposed.z) = transform_samples(f_inv, mc_state.proposed.x)
```